### ISSUE TITLE
Audit Trail plugin support

### ISSUE THREAD

**Issue by qisylvia** [jenkinsci/configuration-as-code-plugin#868]
[jenkins-jira]: https://issues.jenkins-ci.org
[dashboard]: https://issues.jenkins.io/secure/Dashboard.jspa?selectPageId=18341
[contributing]: ../blob/master/docs/CONTRIBUTING.md
[compatibility]: ../blob/master/docs/COMPATIBILITY.md

### Your checklist for this issue

🚨 Please review the [guidelines for contributing][contributing] to this repository.
💡 To better understand plugin compatibility issues, you can [read more here][compatibility]

- [ ] Create an issue on [issues.jenkins-ci.org][jenkins-jira], set the component to the plugin you are reporting it against

- [ ] Before creating an issue on [Jenkins JIRA][jenkins-jira], check for [an existing one via dashboard][dashboard]

- [ ] Link to [Jenkins JIRA issue][jenkins-jira]

- [ ] Ensure [Jenkins JIRA issue][jenkins-jira] has the label `jcasc-compatibility`

- [ ] Link to plugin's GitHub repository

- [x] Link to Plugin Compatibility Tracker #809

### Description

I am trying to use the configuration-as-code plugin to configure Audit Trail plugin:
https://github.com/jenkinsci/audit-trail-plugin

Jenkins version: 2.175
Configuration-as-code plugin version: 1.13
Audit Trail plugin version: 2.4

My audit-trail.yaml file looks like this:
```yml
unclassified:
  auditTrail:
    loggers:
      consoleAuditLogger:
        output: "STD_OUT"
        logPrefix: "AUDIT"
```

When I applied the yaml file using the configuration as code plugin, I get the following error:

Invalid configuration elements for type class jenkins.model.GlobalConfigurationCategory$Unclassified : auditTrail.
Available attributes : administrativeMonitorsConfiguration, artifactManager, buildStepOperation, casCGlobalConfig, cloud, defaultView, extendedEmailPublisher, gitHubConfiguration, gitHubPluginConfig, gitSCM, globalDefaultFlowDurabilityLevel, globalLibraries, location, lockableResourcesManager, mailer, masterBuild, myView, nodeProperties, pipeline-model, plugin, pollSCM, projectNamingStrategy, quietPeriod, scmRetryCount, shell, subversionSCM, timestamperConfig, usageStatistics, viewsTabBar

It might be an usage issue. I read a lot of documentation, but it is not clear what the name should be following "unclassified", for the Audit Trail plugin. I have tried a number of names, such as audit-trail, auditTrailPlugin, etc. 

Based on my reading, most plugins are supported if they followed a specific design pattern. It would be nice if this is documented, or some documentation to help users decide what names should be used for their plugins. Thanks.

**Comment by timja** [comment]
@qisylvia in general if you configure the plugin in the UI, then go to /configuration-as-code and click view configuration, if it doesn't show up there then the plugin isn't supported and needs a fix

Its documented here:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md
and here:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/REQUIREMENTS.md

have you raised an issue on jira for them with the `jcasc-compatibility` label?

**Comment by casz** [comment]
Please follow the checklist and do what it asks of you.

**Comment by casz** [comment]
>Based on my reading, most plugins are supported if they followed a specific design pattern. It would be nice if this is documented, or some documentation to help users decide what names should be used for their plugins. Thanks.

If you open `http://YOUR_JENKINS_URL/configuration-as-code` and hit documentation you get a view of what attributes are configurable on your instance.

**Comment by qisylvia** [comment]
@tim, @casz 
Thank you very much for your quick responses. Much appreciated. 

Prior to opening the issue, I did try both things you suggested: 
1. Open http://YOUR_JENKINS_URL/configuration-as-code and click documentation
2. Go to /configuration-as-code and click view configuration

The audit trail plugin did not show up. Perhaps in my readings I missed the fact that if the plugin does not show up, then it is not supported. Thanks for confirming. 

I opened a Jira with Jenkins to have the plugin updated to support JCasC.
